Code/Framework/AzFramework/Tests/DocumentPropertyEditor/AdapterBuilderTests.cpp

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Tests/DocumentPropertyEditor/DocumentPropertyEditorFixture.h>
+#include <AzFramework/DocumentPropertyEditor/AdapterBuilder.h>
+#include <AzFramework/DocumentPropertyEditor/PropertyEditorNodes.h>
+#include <AzCore/DOM/DomUtils.h>
+#include <AzCore/std/any.h>
+
+namespace AZ::DocumentPropertyEditor::Tests
+{
+    using AdapterBuilderTests = DocumentPropertyEditorTestFixture;
+
+    TEST_F(AdapterBuilderTests, VisitSimpleStructure)
+    {
+        AdapterBuilder builder;
+        builder.BeginAdapter();
+        builder.BeginRow();
+        builder.Label("label");
+        builder.BeginPropertyEditor("TextEditor", Dom::Value("lorem ipsum", true));
+        builder.Attribute(AZ::Name("attr"), Dom::Value(2));
+        builder.EndPropertyEditor();
+        builder.EndRow();
+        builder.EndAdapter();
+
+        Dom::Value result = builder.FinishAndTakeResult();
+
+        /**
+        Expect the following structure:
+        <Adapter>
+            <Row>
+                <Label>label</Label>
+                <PropertyEditor type="TextEditor" attr=2>value</TextEditor>
+            </Row>
+        </Adapter>
+        */
+        Dom::Value expectedValue = Dom::Value::CreateNode("Adapter");
```

Comment:
Should this use the namespace `AZ::DocumentPropertyEditor::Nodes::Adapter::Name` (and similarly below for "Label" and "Row") in case they ever change in the future? I don't imagine they actually would, be just a thought.